### PR TITLE
No list buckets if the key is limited to one

### PR DESCRIPTION
**Pull request by kkalinowski-reef**
- checking whether there is a single bucket allowed for authorised key
- when the name of the bucket is missing, informing user about the issue

### CODE REVIEW

File: b2sdk/api.py [hunk 1]
```diff
@@ -202,6 +202,22 @@ def authorize_account(self, realm, application_key_id, application_key):
...
+            # If we have bucketId set we still need to check bucketName. If the bucketName is None,
+            # it means that the bucketId belongs to a bucket that was already removed.
+            allowed_bucket_name = allowed.get('bucketName')
+            if allowed_bucket_name is not None:
+                self.cache.save_bucket(
+                    self.BUCKET_CLASS(self, allowed_bucket_id, name=allowed_bucket_name)
+                )
+            else:
+                # This key is restricted to a single bucket that doesn't exist.
+                raise RestrictedBucketMissing()
```
Comment (agoncharov-reef) [inline b2sdk/api.py:219]: Do you like "early quit"?

```python
if allowed_bucket_name is None:
    # If we have bucketId set we still need to check bucketName. If the bucketName is None,
    # it means that the bucketId belongs to a bucket that was already removed.
    raise RestrictedBucketMissing()

self.cache.save_bucket(
    self.BUCKET_CLASS(self, allowed_bucket_id, name=allowed_bucket_name)
)
```

File: b2sdk/api.py [hunk 2]
```diff
@@ -202,6 +202,22 @@ def authorize_account(self, realm, application_key_id, application_key):
         """
         self.session.authorize_account(realm, application_key_id, application_key)
 
+        # If the key is restricted to the bucket, pre-populate the cache with it
```
Comment (agoncharov-reef) [inline b2sdk/api.py:205]: I would put it in separate function. Easier to test, modify, and [re]move.

File: b2sdk/exception.py [hunk 3]
```diff
@@ -363,6 +363,14 @@ def __str__(self):
         return 'Application key is restricted to bucket: %s' % self.bucket_name
 
 
+class RestrictedBucketMissing(RestrictedBucket):
+    def __init__(self):
+        super(RestrictedBucketMissing, self).__init__('')
```
Comment (agoncharov-reef) [inline b2sdk/exception.py:368]: `super().__init__('')`

**Comment by kkalinowski-reef** [comment]
Moved to the main repo.

**Pull request closed without merging**